sigmastate/src/main/scala/sigmastate/eval/RuntimeCosting.scala

```diff
@@ -1687,7 +1687,7 @@ trait RuntimeCosting extends CostingRules { IR: IRContext =>
         def tC = evalNode(ctx, env, t)
         def eC = evalNode(ctx, env, e)
         val resV = IF (cC.value) THEN tC.value ELSE eC.value
-        val resCost = opCost(resV, Array(cC.cost, tC.cost, eC.cost), costOf("If", SFunc(Vector(SBoolean, If.tT, If.tT), If.tT)))
+        val resCost = opCost(resV, Array(cC.cost, tC.cost, eC.cost), costOf("If", If.GenericOpType))
```

Comment:
add ScalaDoc for evalNode, better comment  branches inside as well

Reply:
Added ScalaDoc for the method. Note, this entire file will be removed in v5.0.